platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/RepositoryChainArtifactResolver.java

```diff
@@ -77,8 +78,16 @@ private File resolveArtifactLater(ComponentArtifactMetadata artifact, ModuleSour
         return artifactFile.getResult();
     }
 
-    private ModuleComponentRepository<?> findSourceRepository(ModuleSources sources) {
-        RepositoryChainModuleSource repositoryChainModuleSource = sources.getSource(RepositoryChainModuleSource.class).get();
+    private ModuleComponentRepository<?> findSourceRepository(@Nullable ModuleSources sources) {
+        RepositoryChainModuleSource repositoryChainModuleSource = null;
+        if (sources != null) {
+             repositoryChainModuleSource = sources.getSource(RepositoryChainModuleSource.class).orElse(null);
+        }
+
+        if (repositoryChainModuleSource == null) {
+            throw new IllegalArgumentException("No sources provided for artifact resolution");
+        }
```

Comment:
```suggestion
RepositoryChainModuleSource repositoryChainModuleSource;
        if (sources != null) {
             repositoryChainModuleSource = sources.getSource(RepositoryChainModuleSource.class).orElseThrow(() -> new IllegalArgumentException("No sources provided for module for artifact resolution"));
        } else {
            throw new IllegalArgumentException("No module source provided for artifact resolution");
        }
```

🤔 It would be good to distinguish the 2 cases if an exception is thrown.